tests/contrib/celery/test_chained_task.py

```diff
@@ -0,0 +1,46 @@
+import os
+import re
+import signal
+import subprocess
+import time
+
+
+# Ensure that when we call Celery chains, the root span has celery specific span tags
+# The test_integration.py setup doesn't perfectly mimic the condition of a worker process running.
+# This test runs the worker as a side so we can check the tracer logs afterwards to ensure expected span results.
+# See https://github.com/DataDog/dd-trace-py/issues/11479
+def test_task_chain_task_call_task():
+    celery_worker_cmd = "ddtrace-run celery -A tests.contrib.celery.tasks worker -c 1 -l DEBUG -n uniquename1 -P solo"
+    celery_task_runner_cmd = "ddtrace-run python3.11 tests/contrib/celery/run_tasks.py"
+    worker_process = subprocess.Popen(
+        celery_worker_cmd.split(),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        preexec_fn=os.setsid,
+        close_fds=True,
+        cwd=str(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))),
+    )
+
+    # Wait a few seconds for the celery worker to start up before trying to run tasks
+    time.sleep(5)
+    task_runner_process = subprocess.Popen(
+        celery_task_runner_cmd.split(),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        preexec_fn=os.setsid,
+        close_fds=True,
+        cwd=str(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))),
+    )
+
+    task_runner_process.wait()
+
+    # Kill the process so it starts to send traces to the Trace Agent
+    os.killpg(worker_process.pid, signal.SIGKILL)
```

Comment:
why is this needed?

Reply:
If the worker process doesn't get killed, the process will just run endlessly. I'm killing it so there are less logs to parse through.

Reply:
ok - this won't work on windows, though (see https://stackoverflow.com/questions/28551180/how-to-kill-subprocess-python-in-windows)

Reply:
It looks like we call `os.killpg` in several places today: https://github.com/search?q=repo%3ADataDog%2Fdd-trace-py%20os.killpg&type=code

Is there a way to exclude some tests from running on Windows?

Reply:
Update: I think I can achieve this with just the subprocess instead: https://docs.python.org/3/library/subprocess.html#subprocess.Popen.kill, which should make it more stable in windows.

I'll see how it performs in CI though.

Reply:
the tests pass in CI now :)